src/Npgsql/NpgsqlTypes/NpgsqlTypes.cs

```diff
@@ -469,6 +469,13 @@ public static explicit operator IPAddress(NpgsqlInet inet)
     public static implicit operator NpgsqlInet(IPAddress ip)
         => new(ip);
 
+    public static implicit operator NpgsqlInet(IPNetwork cidr)
```

Comment:
This mirrors the implicit cast we already had from the now-obsolete NpgsqlCidr (which was replaced by IPNetwork).